cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -191,8 +193,15 @@ struct Divide {
   template <typename T, typename Arg0, typename Arg1>
   static enable_if_floating_point<T> Call(KernelContext* ctx, Arg0 left, Arg1 right) {
     if (ARROW_PREDICT_FALSE(right == 0)) {
```

Comment:
This condition shouldn't be necessary anymore. You should be able to call [`feholdexcept`](https://en.cppreference.com/w/cpp/numeric/fenv/feholdexcept) at the beginning of the kernel, so that the FPU does the right thing. Also, restore the previous FP environment using `fesetexcept` at the end.

Reply:
That's awesome! Thank you for the good suggestion.

Reply:
After changing the code, the AMD64 Ubuntu 18.04 C++ ASAN UBSAN integration test is failing again. 
It seems this is a bug of clang, and some special flags are required:

https://bugs.llvm.org/show_bug.cgi?id=17000#c1

Reply:
@pitrou Thanks for your effort. It is working like a charm!